diff --git a/bin/coditz.rb b/bin/coditz.rb
new file mode 100755
index 0000000..739ba8e
--- /dev/null
+++ b/bin/coditz.rb
@@ -0,0 +1,53 @@
+#!/usr/bin/ruby
+$LOAD_PATH.unshift(File.expand_path("#{File.dirname($0)}/../lib"))
+require 'rubygems'
+require 'cinatra'
+#require 'yaml_waml'
+
+# FIXME see http://github.com/jugyo/cinatra/issues/#issue/4
+begin
+
+require 'ditz'
+
+CONFIG_FN = ".ditz-config"
+PLUGIN_FN = ".ditz-plugins"
+
+=begin
+bin/ditz
+lib/ditz.rb
+lib/ditz/operator.rb
+lib/ditz/views.rb
+lib/ditz/model-objects.rb
+lib/ditz/model.rb
+lib/ditz/lowline.rb
+=end
+
+def ditz_op(cmd, opts = [])
+  config_dir = Ditz::find_dir_containing CONFIG_FN
+  plugin_dir = Ditz::find_dir_containing PLUGIN_FN
+  op = Ditz::Operator.new
+  config = Ditz::Config.from File.join(config_dir || ".", CONFIG_FN)
+  config.use_editor_if_possible = false # overwrite
+  config.paginate = "never" # overwrite
+  issue_dir = Pathname.new(config.issue_dir)
+  storage = Ditz::FileStorage.new issue_dir
+  project = storage.load
+  op.do cmd, project, config, opts
+end
+
+command 'dhelp' do |arg|
+  ditz_op 'help'
+end
+
+command 'print_hooks' do |arg|
+  Ditz::HookManager.print_hooks
+end
+
+command 'todo' do |arg|
+  ditz_op 'todo'
+end
+
+# FIXME see http://github.com/jugyo/cinatra/issues/#issue/4
+rescue => x
+  p x
+end
