diff --git a/lib/ditz/model-objects.rb b/lib/ditz/model-objects.rb
index 91234c3..98df248 100644
--- a/lib/ditz/model-objects.rb
+++ b/lib/ditz/model-objects.rb
@@ -166,7 +166,8 @@ class Issue < ModelObject
   attr_accessor :name, :pathname, :project
 
   ## these are the fields we interpolate issue names on
-  INTERPOLATED_FIELDS = [:title, :desc, :log_events]
+  #INTERPOLATED_FIELDS = [:title, :desc, :log_events]
+  INTERPOLATED_FIELDS = []
 
   STATUS_SORT_ORDER = { :unstarted => 2, :paused => 1, :in_progress => 0, :closed => 3 }
   STATUS_WIDGET = { :unstarted => "_", :in_progress => ">", :paused => "=", :closed => "x" }
