diff --git a/lib/ditz/views.rb b/lib/ditz/views.rb
index f1d9da9..e41c9e9 100644
--- a/lib/ditz/views.rb
+++ b/lib/ditz/views.rb
@@ -161,9 +161,9 @@ EOS
           # id
           f.print ":#{issue.id} a "
           f.print case issue.type 
-                when :bugfix, :bug: "baetle:Bug"
-                when :feature: "baetle:Enhancement"
-                when :task: "wf:Task"
+                when :bugfix, :bug; "baetle:Bug"
+                when :feature; "baetle:Enhancement"
+                when :task; "wf:Task"
                 end
           f.puts " ;"
           # title
@@ -173,10 +173,10 @@ EOS
           # state
           f.print "    wf:state baetle:" 
           f.print case issue.status
-                  when :unstarted: "New"
-                  when :in_progress: "Started"
-                  when :closed: "Closed"
-                  when :paused: "Later"
+                  when :unstarted; "New"
+                  when :in_progress; "Started"
+                  when :closed; "Closed"
+                  when :paused; "Later"
                   end
           f.puts " ;"
           # created
