diff --git a/bin/ditzweb b/bin/ditzweb
index ba0419f..40df676 100755
--- a/bin/ditzweb
+++ b/bin/ditzweb
@@ -191,6 +191,14 @@ post '/i/?' do
   redirect "/i/#{issue.id}"
 end
 
+get '/repo/?' do
+  # TODO git support
+  puts `hg addremove`
+  puts `hg ci -m save`
+  puts `hg push`
+  redirect '/'
+end
+
 require 'optparse'
 OptionParser.new { |op|
   op.on('-x')        {       set :lock, true }
@@ -200,4 +208,4 @@ OptionParser.new { |op|
   op.on('-o addr')   { |val| set :bind, val }
 }.parse!(ARGV.dup)
 
-Sinatra::Application.run!
\ No newline at end of file
+Sinatra::Application.run!
diff --git a/contrib/tombloo_ditz.js b/contrib/tombloo_ditz.js
new file mode 100644
index 0000000..32100b0
--- /dev/null
+++ b/contrib/tombloo_ditz.js
@@ -0,0 +1,32 @@
+// ../../extensions/
+models.register({
+	name : 'Ditz',
+	ICON : 'chrome://tombloo/skin/item.ico',
+	check : function(ps){
+		return (/(regular|photo|quote|link|conversation|video)/).test(ps.type) && !ps.file;
+	},
+
+	getToken : function(){
+		return getCookieString('localhost', 'AT').split('=').pop();
+	},
+
+	post : function(ps){
+		var self = this;
+		return request('http://localhost:4567/i/', {
+			redirectionLimit : 0,
+			sendContent : {
+//			at      : self.getToken(),
+//				tags       : joinText(ps.tags, ','),
+//				local      : this.getLocalTimestamp(),
+//				private    : ps.private? 1 : 0,
+				summary : ps.page,
+				details : joinText([ps.body, ps.description], ' ', true),
+				reference : ps.pageUrl,
+				image0  : ps.type == 'photo' ? ps.itemUrl : '',
+//      type : '',
+//      release : '',
+//      component : '',
+			},
+		});
+	},
+});
diff --git a/lib/views/index.haml b/lib/views/index.haml
index 166d353..95bb473 100644
--- a/lib/views/index.haml
+++ b/lib/views/index.haml
@@ -21,6 +21,8 @@
         %input#ignorecase{:type => :checkbox, :name => :i, :checked => true}
         %label{:for => :ignorecase} Ignore case
         %input{:type => :submit, :value => 'Search'}
+      %form{:action => '/repo', :method => :get}
+        %input{:type => :submit, :value => 'hg addremove ; hg commit -m save ; hg push'}
       %a{:href => "javascript:var%20d=document,w=window,e=w.getSelection,k=d.getSelection,x=d.selection,s=(e?e():(k)?k():(x?x.createRange().text:0)),f='http://localhost:4567/i/new',l=d.location,e=encodeURIComponent,p='?u='+e(l.href)+'&t='+e(d.title)+'&s='+e(s),u=f+p;l.href=u;"} Bookmarklet
       - releases.each do |key,release|
         %h2= h release[:title]
diff --git a/sheila/sheila.rb b/sheila/sheila.rb
index 8e40f34..6d525fb 100644
--- a/sheila/sheila.rb
+++ b/sheila/sheila.rb
@@ -16,7 +16,7 @@ class String
   def gravatar(s=20)
     email = split.last
     email = email[1, email.size - 2] if email[0, 1] == '<'
-    "http://www.gravatar.com/avatar/#{Digest::MD5.hexdigest(email)}?s=#{s}"
+    "http://www.gravatar.com/avatar/#{Digest::MD5.hexdigest(email)}?s=#{s}&default=identicon"
   end
 end
 
